### PR TITLE
Add validation for VK_KHR_pipeline_executable_properties 

### PR DESCRIPTION
**Pull request by jekstrand**

### CODE REVIEW

**Comment by jekstrand** [comment]
This PR is based on #1096 at the moment.  I'll rebase once a header update happens in master regardless of whether #1096 gets merged or someone at LunarG does the update via some other process.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -5271,6 +5277,83 @@ void CoreChecks::PostCallRecordCreateRayTracingPipelinesNV(VkDevice device, VkPi
...
+    if (!enabled_features.pipeline_exe_props_features.pipelineExecutableInfo) {
+        skip |= log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT, HandleToUint64(device),
+                        "VUID-vkGetPipelineExecutablePropertiesKHR-pipelineExecutableProperties-03270",
+                        "vkGetPipelineExecutablePropertiesKHR called when pipelineExecutableInfo feature is not enabled.");
+    }
+
+    return skip;
+}
+
+bool CoreChecks::ValidatePipelineExecutableInfo(VkDevice device, const VkPipelineExecutableInfoKHR *pExecutableInfo) {
```
Comment (mark-lunarg) [inline layers/core_validation.cpp]: @jekstrand, local validation helper functions need to be const.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -5271,6 +5277,83 @@ void CoreChecks::PostCallRecordCreateRayTracingPipelinesNV(VkDevice device, VkPi
...
+    return skip;
+}
+
+bool CoreChecks::PreCallValidateGetPipelineExecutableStatisticsKHR(VkDevice device,
+                                                                   const VkPipelineExecutableInfoKHR *pExecutableInfo,
+                                                                   uint32_t *pStatisticCount,
+                                                                   VkPipelineExecutableStatisticKHR *pStatistics) {
+    bool skip = ValidatePipelineExecutableInfo(device, pExecutableInfo);
+
+    PIPELINE_STATE *pipeline_state = GetPipelineState(pExecutableInfo->pipeline);
```
[the remark this answers is not in the document]
Comment (jekstrand) [reply layers/core_validation.cpp]: Done.  Both this instance and the one below in `PreCallValidateGetPipelineExecutableInternalRepresentationsKHR`

File: layers/core_validation.cpp (hunk 1 above)
Comment (jekstrand) [reply layers/core_validation.cpp]: Done, assuming this is the only one that counts as a "local validation helper"

**Comment by mark-lunarg** [comment]
Rebased and running through CI.

**Pull request merged**